mantidproject/mantid
# Zoom Monitor 4 Movement

**Description of work.**
Adds the ability to move monitor 4 (in addition to 5) on Zoom:
- Updates the GUI to add a new box (disabled on all but Zoom)
- Updates the parser to understand the new command `TRANS/SHIFT=4000 5` where 4000 is distance and 5 is the monitor

Refactoring to add supporting unit tests:
- Everything related to Settings->Adjustment from `run_tab_presenter` and `state_gui_model` was moved into `settings_adjustment_presenter` and `settings_adjustment_model` respectively
- Moved common elements into `model_common` and `presenter_common`

N.B. Everything after `settings_adjustment_presenter:42` and `settings_adjustment_model:39` is a cut + paste (and minor renaming)

The wiki has been [updated](https://www.mantidproject.org/SANS_User_File_Commands#TRANS)

**Report to:** Dominic

**To test:**
- Open the ISIS SANS GUI
- Use a SANS2D file and go to settings -> adjustment. Ensure box is greyed out for monitor 5
- Use a ZOOM file and ensure the same box is enabled
- Attempt to run a reduction using ZOOM dataset `6113` (which is an empty) using monitor 4 and 5

- Modify the Zoom user file to contain the following
```
TRANS/SHIFT=4000 4
TRANS/SHIFT=5000 5
TRANS/SHIFT=6000 4
```
- Ensure monitor 4 is set to 6000 in the GUI (last value) and monitor 5 is set to 5000
- Modify the Zoom user file to contain the following
```
TRANS/SHIFT=4000 3
TRANS/SHIFT=5000 5
```
- Ensure this is not accepted (no monitor 3)
- Open a SANS2D user file and check the shift matches the one specified in the file
- Insert the following into the SANS2D file `TRANS/SHIFT=6000 4`
- Check that this also changes monitor 4 to 6000

Fixes #26375 . 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
